src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -0,0 +1,194 @@
+package nextstep.subway.domain;
+
+import nextstep.subway.exception.IllegalUpdatingStateException;
+
+import javax.persistence.*;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Embeddable
+public class Sections {
+    public static final int MIN_SECTION_SIZE = 1;
+
+    @OneToMany(mappedBy = "line",
+            fetch = FetchType.LAZY,
+            cascade = {CascadeType.PERSIST, CascadeType.MERGE},
+            orphanRemoval = true)
+    private List<Section> sections = new ArrayList<>();
+
+    public Sections() {
+    }
+
+    public Sections(Section section) {
+        this.sections.add(section);
+    }
+
+    public void addSection(Section section) {
+        if (sections.isEmpty()) {
+            sections.add(section);
+            return;
+        }
+        checkStateToAddSection(section);
+
+        if (addSectionBetweenStations(section)) {
+            return;
+        }
+        if (addSectionInFrontOfFirstUpStation(section)) {
+            return;
+        }
+        addSectionBehindLastDownStation(section);
+    }
+
+    private void checkStateToAddSection(Section section) {
+        List<Station> allStations = getAllStations();
+        if (allStations.contains(section.getUpStation()) && allStations.contains(section.getDownStation())) {
+            throw new IllegalUpdatingStateException("요청한 구간의 상행역과 하행역 모두 이미 노선에 등록되어있어 등록이 불가합니다.");
+        }
+        if (!allStations.contains(section.getUpStation()) && !allStations.contains(section.getDownStation())) {
+            throw new IllegalUpdatingStateException("요청한 구간의 상행역과 하행역 모두 기존 노선에 등록되어 있지않아 등록이 불가합니다.");
+        }
+    }
+
+    private boolean addSectionBetweenStations(Section section) {
+        if (addSectionBetweenStationsWhenSameUpStation(section)) {
+            return true;
+        }
+        if (addSectionBetweenStationsWhenSameDownStation(section)) {
+            return true;
+        }
+        return false;
+    }
```

Comment:
다음과 같이 해볼 수 있을 것 같아요 😄 
```suggestion
    private boolean addSectionBetweenStations(Section section) {
        if (addSectionBetweenStationsWhenSameUpStation(section)) {
            return true;
        }
        return addSectionBetweenStationsWhenSameDownStation(section);
    }```